client/components/admin/includes/add-book.js

```diff
@@ -0,0 +1,222 @@
+                    return false;
+                    break;
+                }
+            case 'description':
+                if (value.length < 5 || !value) {
+                    this.setState({ descError: 'Book description is required' })
+                    break;
+                }
+            case 'isbn':
+                if (value.length < 5 || !value) {
+                    this.setState({ isbnError: 'Book ISBN must be a minimum of 5 characters' })
+                }
+        }
+    }
+
+    handleSubmit(e) {
+        e.preventDefault();
+        console.log(this.state);
+        this.props.onSubmit(this.state)
+            .then((message) => {
+                Materialize.toast('Book added Successfully', 2000, 'blue',
+                    () => {
+                        this.setState({ isLoading: false })
+                    });
+                window.location.href = '/admin';
+            })
+            .catch((err) => err)
+    }
+
+    render() {
+        return (
+            <div style={{ marginTop: 20, backgroundColor: '#fff', width: '70%', float: 'right', marginRight: 100 }}>
+                <div className="row">
+                    <form name="add_book" className="col s12" onSubmit={this.handleSubmit}>
+                        <div className="add-book">
+                            <div className="row">
+                                <div className="input-field col s12">
+                                    <input
+                                        id="title"
+                                        type="text"
+                                        name="title"
+                                        className="validate"
+                                        onChange={this.onChange}
+                                        onFocus={this.onFocus}
+                                        onBlur={this.onBlur}
+                                        required />
+                                    <label htmlFor="isbn">Title</label>
+                                    <div style={{ color: 'red' }}>{this.state.titleError}</div>
+                                </div>
+                            </div>
+                            <div className="row">
+                                <div className="input-field col s12">
+                                    <input
+                                        id="author"
+                                        type="text"
+                                        name="author"
+                                        className="validate"
+                                        onChange={this.onChange}
+                                        onFocus={this.onFocus}
+                                        onBlur={this.onBlur}
+                                        required />
+                                    <label htmlFor="isbn">Author</label>
+                                    <div style={{ color: 'red' }}>{this.state.authorError}</div>
```

Comment:
DRY - Don't repeat yourself
You are repeating same color twice. Try to make the style an object so that you don't have to pass the color twice.

Reply:
Thanks for the feedback, I will surely fix it up.